lib/pbench/server/api/resources/upload_api.py

```diff
@@ -76,7 +77,15 @@ def __init__(self, config: PbenchServerConfig):
                 ApiMethod.PUT,
                 OperationCode.CREATE,
                 uri_schema=Schema(Parameter("filename", ParamType.STRING)),
-                query_schema=Schema(Parameter("access", ParamType.ACCESS)),
+                query_schema=Schema(
+                    Parameter("access", ParamType.ACCESS),
+                    Parameter(
+                        "metadata",
+                        ParamType.LIST,
+                        element_type=ParamType.STRING,
+                        string_list=",",
+                    ),
```

Comment:
So this will enable to add metadata `archiveonly` :+1:

Reply:
In theory, we could do that as metadata, but we'd have to designate a `global` sub-namespace that the client can set. My idea was to do it as a separate query parameter, and `PUT` could record it on a `server.archiveonly` or some such, which would be visible but not so easily changeable.

But, yeah, really we could treat `server.archiveonly` specially, like `server.deletion`, allow it to be set/changed, and even `READY` the dataset for `UNPACK` if it's changed later to `True`. Maybe? 😕

Reply:
We could, but it wouldn't really make much difference because I want to carefully validate (and often transform) what I allow users to set. Although the segregation could be documented, the individual specific keywords still need to be explicitly and individually documented anyway.